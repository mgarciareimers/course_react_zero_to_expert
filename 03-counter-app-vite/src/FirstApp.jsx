import PropTypes from 'prop-types';

const sendMessage = (user, text, age) => {
    return {
        'user': user,
        'text': text,
        'age': age,
    };
}

const FirstApp = ({ user, text, age }) => {
    const newMessage = sendMessage(user, text, age);

    return (
        <>
            <b data-testid='test-title'>{ newMessage.user } ({ newMessage.age }):</b>
            <p>{ newMessage.text }</p>
        </>
    );
}

FirstApp.propTypes = {
    user: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    age: PropTypes.number.isRequired,
}

// Not mandatory.
FirstApp.defaultProps = {
    text: 'No text',
    age: 0
}

export default FirstApp;