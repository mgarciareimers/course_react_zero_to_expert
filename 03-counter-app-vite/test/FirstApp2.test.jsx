import FirstApp from '../src/FirstApp';
import { getAllByAltText, render, screen } from '@testing-library/react';

describe('Tests of <FirstApp />>', () => { 
    const user = 'Mike';
    const age = 10;
    const title = `${ user } (${ age }):`;
    const text = 'Hello World!';

    test('Match with snapshot', () => {
        const { container } = render(
            <FirstApp 
                user={ title }
            />
        );

        expect(container).toMatchSnapshot();
    })

    // Second test.
    test('Title has to be', () => {
        render(
            <FirstApp 
                user={ user }
                age={ age }
            />
        );
        
        expect(screen.getByText(title)).toBeTruthy();
    });

    // Third test.
    test('Title has to be in b', () => {
        render(
            <FirstApp 
                user={ user }
                age={ age }
            />
        );
        
        expect(screen.getByRole('paragraph')).toBeTruthy();
    });

    // Third test.
    test('Show text sent by props', () => {
        const { getAllByText } = render(<FirstApp 
            user={ user }
            age={ age }
            text={ text }
        />);
        
        expect(screen.getAllByText(text).length).toBe(1);
    });
});