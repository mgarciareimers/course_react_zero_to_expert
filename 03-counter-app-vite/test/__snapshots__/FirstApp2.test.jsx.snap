// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <FirstApp />> Match with snapshot 1`] = `
<div>
  <b
    data-testid="test-title"
  >
    Mike (10):
     (
    0
    ):
  </b>
  <p>
    No text
  </p>
</div>
`;
